### ISSUE TITLE
feat: Prevent suspend during presentation w/ Screen Wake Lock API (crosspost)

### ISSUE THREAD

**Issue by batonac** [ebullient/obsidian-slides-extended#5]
Currently the screen can turn off and the computer suspend during a Advanced Slides presentation, which is problematic. The computer can be set to disable suspend, but that's a cumbersome solution.

It seems easy enough to utilize the [Screen Wake Lock API](https://developer.mozilla.org/en-US/docs/Web/API/Screen_Wake_Lock_API) to prevent suspend while the presentation is active:
```js
// Create a reference for the Wake Lock.
let wakeLock = null;

// create an async function to request a wake lock
try {
  wakeLock = await navigator.wakeLock.request("screen");
  statusElem.textContent = "Wake Lock is active!";
} catch (err) {
  // The Wake Lock request has failed - usually system related, such as battery.
  statusElem.textContent = `${err.name}, ${err.message}`;
}
```
[Secure context](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts) is required, but apparently localhost qualifies:
> Locally-delivered resources such as those with http://127.0.0.1 URLs, http://localhost and http://*.localhost URLs (e.g. http://dev.whatever.localhost/), and file:// URLs are also considered to have been delivered securely.

I'm not sure of the best way to differentiate slide "preview" from "presentation mode". The easiest way would be to ensure that the Wake Lock isn't active unless the slides are viewed in an external browser. A more sophisticated approach would be request a wake lock only when the browser window enters full screen mode, and release it when exiting full screen mode.

**Comment by batonac** [comment]
This is a cross post from my original post here: https://github.com/MSzturc/obsidian-advanced-slides/issues/296

@ebullient thanks for keeping this software alive!

**Comment by ebullient** [comment]
Try a PR? Looks like you're pretty close. Feels like this should go in the reveal.html template (rather than embed?) .. I did add a function in there to detect iframe usage for other reasons..

**Comment by batonac** [comment]
I haven't been able to get this working in local dev yet 🤔

**Comment by ebullient** [comment]
No hurry. It isn't something I use, so I'd rather you get it working so it works for you, and then we'll figure out how to make sure it ages well. ;)

**Comment by ebullient** [comment]
Just bumping this to remind you that I'm waiting for you. ;)

**Comment by batonac** [comment]
Yes, sorry. I'm buried with other things at the moment, but hoping to return to this soon.
